### PR TITLE
Added basic bandwidth accounting experiment

### PR DESCRIPTION
**Pull request by devos50**
This PR adds a basic bandwidth accounting experiment, following the merge of https://github.com/Tribler/tribler/pull/5626 in Tribler. The experiment initiates random bandwidth payouts with random users, and afterwards aggregates all payouts in a single database.

The validation experiment can be found [here](https://jenkins-ci.tribler.org/job/validation_experiments/job/validation_experiment_payouts/).

### CODE REVIEW

File: experiments/bandwidth_accounting/bandwidth_accounting_module.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            self.payouts_task = run_task(self.do_random_payout, interval=1.0)
+
+        run_task(start, delay=delay)
+
+    def do_random_payout(self):
+        verified_peers = list(self.overlay.network.verified_peers)
+        random_peer = random.choice(verified_peers)
+        random_amount = random.randint(1 * 1024 * 1024, 10 * 1024 * 1024)
+        self._logger.info("Performing payout to peer %s with amount %d", random_peer, random_amount)
+        ensure_future(self.overlay.do_payout(random_peer, random_amount))
```
Comment (qstokkink) [inline experiments/bandwidth_accounting/bandwidth_accounting_module.py]: Nitpicking: this works, but why not register the future somewhere if you go through the trouble of ensuring it?

File: experiments/bandwidth_accounting/post_process_bandwidth_accounting.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+        with open("total_payout_edges.txt", "w") as out_file:
+            out_file.write("%d" % len(transactions))
+
+    def run(self):
+        self.aggregate_databases()
+
+
+# cd to the output directory
+os.chdir(os.environ['OUTPUT_DIR'])
```
Comment (qstokkink) [inline experiments/bandwidth_accounting/post_process_bandwidth_accounting.py:48]: This is very error-prone. I would suggest stripping this and using `with open(os.path.join(os.environ['OUTPUT_DIR'], "total_payout_edges.txt"), "w") as out_file:` on line 40 instead.

I assume that is the only reason this `chdir` is here, please let me know if that is not the case and I will update my review.

**Review by qstokkink — CHANGES_REQUESTED** [review]

File: experiments/bandwidth_accounting/post_process_bandwidth_accounting.py (hunk 2 above)
Comment (devos50) [reply experiments/bandwidth_accounting/post_process_bandwidth_accounting.py:48]: I had to dig a bit deeper into this, but it seems that the argument passed to the init method of `BandwidthStatisticsParser` is `.`. Therefore, this class makes the hidden assumption that the current working directory is in the experiment output directory, which is definitely not a good design. However, this code has been in Gumby for many, many years now. Fixing this is possible but would require refactoring efforts unrelated to this PR. Therefore, I opt to leave it like this for now. 👍

File: experiments/bandwidth_accounting/bandwidth_accounting_module.py (hunk 1 above)
Comment (devos50) [reply experiments/bandwidth_accounting/bandwidth_accounting_module.py]: True, that was unnecessary. I removed the `ensure_future`.

File: experiments/bandwidth_accounting/post_process_bandwidth_accounting.py (hunk 2 above)
Comment (qstokkink) [reply experiments/bandwidth_accounting/post_process_bandwidth_accounting.py:48]: Seems reasonable, let's defer that to an issue.

**Review by qstokkink — APPROVED** [review]

**Pull request merged**